qiskit/providers/aer/pulse_new/models/signals.py

```diff
@@ -374,3 +374,106 @@ def signal_add(sig1: Union[BaseSignal, float, int, complex],
 
     # Other symmetric cases
     return signal_add(sig2, sig1)
+
+
+class VectorSignal:
+    """The vector version of the Signal class - the envelope is an array-valued
+    function, and carrier_freqs is an array of carrier frequencies.
+
+    In addition, a drift_array is set to correspond to the value of the
+    VectorSignal when all "time-dependent terms" are off. E.g. if it is
+    composed of a list of Signal objects, this corresponds to the output when
+    all non-Constant signal objects are zero.
+    """
+
+    def __init__(self,
+                 envelope: Callable,
+                 carrier_freqs: np.array,
+                 drift_array: np.array = None):
+        """Initialize with vector-valued envelope, carrier frequencies for
+        each entry, and a drift_array, which corresponds to the value of the
+        signal when the "time-dependent terms" are "off".
+
+        Args:
+            envelope: function of a single float returning an array.
+            carrier_freqs: list of carrier frequences for each component of the
+                           envelope.
+            drift_array: a default array meant to be the value of the envelope
+                         when all "time-dependent terms" are off.
+        """
+        self.envelope = envelope
+        self.carrier_freqs = carrier_freqs
+
+        # if not supplied nothing is assumed, constant array is taken as all
+        # zeros
+        if drift_array is None:
+            self.drift_array = np.zeros(len(self.carrier_freqs))
+        else:
+            self.drift_array = drift_array
+
+    @classmethod
+    def from_signal_list(cls, signal_list: List[Signal]):
```

Comment:
Can we initialize from `BaseSignal` instead? `BaseSignal` defines an `envelop_value` function so this should be fine.
```suggestion
    def from_signal_list(cls, signal_list: List[BaseSignal]):
```